File: frontend/src/models/messages/AnswerMessage.ts

import { MessageType } from "./MessageType";
import BaseMessage from "./BaseMessage";
import { AnswerMessageData } from "./AnswerMessageData";
import { UUID } from "../shared/UUIDType";

/**
 * Representa un mensaje de respuesta, extendiendo la funcionalidad de BaseMessage.
 */
export default class AnswerMessage extends BaseMessage {
    
    /** Tipo del mensaje, en este caso, una respuesta. */
    type: MessageType = MessageType.ANSWER;
    
    /** Contenido del mensaje, que contiene la respuesta. */
    content: AnswerMessageData = { text: "" };

    /**
     * Crea una nueva instancia de AnswerMessage.
     * @param id - Identificador único del mensaje.
     * @param content - Contenido del mensaje con la respuesta.
     */
    constructor(id: UUID, content: AnswerMessageData) {
        super(id);
        this.content = content;
    }

    /**
     * Obtiene el identificador del mensaje.
     * @returns El UUID del mensaje.
     */
    getId(): UUID {
        return this.id;
    }
    
    /**
     * Obtiene el tipo del mensaje.
     * @returns El tipo del mensaje, en este caso, MessageType.ANSWER.
     */
    getType(): MessageType {
        return this.type;
    }

    /**
     * Obtiene el contenido del mensaje.
     * @returns El contenido del mensaje, de tipo AnswerMessageData.
     */
    getContent<T = AnswerMessageData>(): T {
        return this.content as T;
    }
}

File: api-python/app/messages/AnswerMessage.py
from pydantic import BaseModel
from typing import List
from uuid import UUID
from .BaseMessage import BaseMessage
from .MessageType import MessageType

class AnswerMessageContent(BaseModel):
    """
    Contenido de un mensaje de respuesta.

    Esta clase define el contenido específico de un mensaje de respuesta, 
    incluyendo el ID del chat, el ID del mensaje y el texto de la respuesta.
    """

    text: str
    """
    Texto de la respuesta.

    Este campo contiene el texto que será enviado como parte de la respuesta al usuario.
    """

class AnswerMessage(BaseMessage[AnswerMessageContent]):
    """
    Mensaje de respuesta.

    Esta clase extiende la clase base `BaseMessage` y agrega un contenido específico de respuesta.
    Además, incluye una marca de tiempo y un tipo de mensaje predeterminado (ANSWER).
    """

    timeStamp: int
    """
    Marca de tiempo del mensaje de respuesta.

    Este campo contiene el tiempo en formato de timestamp (segundos desde la época Unix) 
    en que se generó la respuesta.
    """

    type: MessageType = MessageType.ANSWER
    """
    Tipo de mensaje.

    Este campo indica el tipo de mensaje. En este caso, siempre tiene el valor `ANSWER`, 
    ya que esta clase se utiliza para mensajes de respuesta.
    """
